src/ConstantBounds.cpp

```diff
@@ -0,0 +1,1144 @@
+#include "ConstantBounds.h"
+#include "Bounds.h"
+#include "Error.h"
+#include "IR.h"
+#include "IREquality.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "IRVisitor.h"
+#include "Simplify.h"
+#include "SimplifyCorrelatedDifferences.h"
+#include "UniquifyVariableNames.h"
+
+#include <vector>
+
+namespace Halide {
+namespace Internal {
+
+static bool disable_approximate_methods() {
+    return get_env_variable("HL_DISABLE_APPROX_CBOUNDS") == "1";
```

Comment:
Do we want to pay this cost for every call to `find_constant_bound[s]`? Is this just for debugging or for "real" cases? Maybe we should instead do something like

```
static bool disable_approximate_methods() {
    static bool b = get_env_variable("HL_DISABLE_APPROX_CBOUNDS") == "1";
    return b;
}
```

Reply:
Oops, good catch, thanks! Will fix.

Reply:
Actually, I'm not even entirely sure this should remain here, I considered allowing the new methods to be toggled on or off, but I'm not sure we really need that. @abadams what do you think?

Reply:
If you leave this in then someone, somewhere absolutely will end up basing an entire industry on this one little env var, so unless there is a crystal clear use for it now and in the future, please remove it, or at least leave commented-out by default.

Reply:
Removed in 6577f30